src/builders.rs

```diff
@@ -1230,3 +1233,129 @@ impl From<DateBased> for Output {
+    fn default() -> Self {
+        FormatterBuilder {
+            #[cfg(feature = "colored")]
+            color_config: Some(Default::default()),
+            #[cfg(feature = "chrono")]
+            chrono: true,
+            level: true,
+            target: true,
+        }
+    }
+}
+
+impl FormatterBuilder {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn level(mut self, level: bool) -> Self {
+        self.level = level;
+        self
+    }
+
+    pub fn target(mut self, target: bool) -> Self {
+        self.target = target;
+        self
+    }
+
+    #[cfg(feature = "colored")]
+    pub fn color(mut self, color: bool) -> Self {
+        self.color_config = if color {
+            self.color_config.or_else(Default::default)
+        } else {
+            None
+        };
+        self
+    }
+
+    #[cfg(feature = "colored")]
+    pub fn color_config(
+        mut self,
+        modify_config: impl FnOnce(ColoredLevelConfig) -> ColoredLevelConfig,
+    ) -> Self {
+        self.color_config = self.color_config.map(modify_config);
+        self
+    }
+
+    #[cfg(feature = "chrono")]
+    pub fn chrono(mut self, chrono: bool) -> Self {
+        self.chrono = chrono;
+        self
+    }
+
+    #[rustfmt::skip]
+    pub fn build(
+        self,
+    ) -> impl Fn(FormatCallback<'_>, &std::fmt::Arguments<'_>, &log::Record<'_>) + Sync + Send + 'static
+    {
+        move |out, message, record| {
+            let time = {
+                #[cfg(feature = "chrono")]
+                {
+                    self.chrono
+                        .then(|| chrono::Local::now().format("%Y-%m-%d %H:%M:%S,%3f"))
+                }
+                #[cfg(not(feature = "chrono"))]
+                {
+                    None
+                }
+            };
+
+            let level = if self.level {
+                #[cfg(feature = "colored")]
+                {
+                    Some(
+                        self.color_config
+                            .map(|config| config.color(record.level()))
+                            .unwrap_or_else(|| crate::colors::WithFgColor {
+                                text: record.level(),
+                                color: None,
+                            }),
+                    )
+                }
+                #[cfg(not(feature = "colored"))]
+                {
+                    Some(record.level())
+                }
+            } else {
+                None
+            };
+
+            let target = self.target.then(|| {
+                if !record.target().is_empty() {
+                    record.target()
+                } else {
+                    record.module_path().unwrap_or_default()
+                }
+            });
+
+            /* Sadly we cannot store and compose std::fmt::Arguments due to lifetime issues.
+             * In order to avoid unnecessary write calls nevertheless, we must enumerate all options
+             */
```

Comment:
Have you tried composing them using nested matches, with no temporary variables? I think something like this _should_ work:

```
format_args!("{}{}", match time { ... => format_args!(...) }, match level { ... => format_args!(...) })
```

At least in the past, I've never been able to store the result of `format_args!` in a variable, but it seems to always work to let it live temporarily while being passed to another function. That's the origin of `out.finish`'s design, too.

Reply:
Sadly, no. As soon as I wrap the inner `format_args!` in an expression, the error message appears. To be honest, this is really frustrating and I'll open an issue to change this in Rust because according to a quick search we're not the only ones running into this.